dimod/core/initialized.py

```diff
@@ -47,31 +52,28 @@ class Initialized(abc.ABC):
 
     # dev note: if this function is updated, make sure to also update
     # IdentitySampler
-    def parse_initial_states(self, bqm,
-                             initial_states=None,
-                             initial_states_generator='random',
-                             num_reads=None, seed=None, copy_always=False):
-        """Parses/generates initial states for an initialized sampler.
+    def parse_initial_states(self, bqm: BinaryQuadraticModel,
+                             initial_states = None,
+                             initial_states_generator: str = 'random',
```

Comment:
If we want to be fancy, we should use `typing.Literal`, but since that arrived in Python 3.8 we could do at the top
```python
try:
    InitialStateGenerators = typing.Literal['random', ...
except ImportError:
    InitialStateGenerators = str
```

Reply:
@arcondello I think for this to not fail CI Py 3.7 it needs also something like this:
```
from typing import TYPE_CHECKING
if TYPE_CHECKING:             # not entered at runtime
    from typing import Literal
```

Reply:
`from typing import Literal` will fail in 3.7 always. It will just silently fail when mypy runs in `TYPE_CHECKING` mode
the `if TYPE_CHECKING:` guard only affects runtime vs not.

So you can do
```python
if TYPE_CHECKING:
    try:
        InitialStateGenerators = typing.Literal['random', ...
    except ImportError:
        InitialStateGenerators = str
```
if you want to do both but :shrug: IMO that's overkill.

Reply:
Correction, you need `except AttributeError:`.